routes/user.js

```diff
@@ -1,9 +1,21 @@
 const { Router } = require("express");
 const router = Router();
 
+// Middlewares
+const authenticateUser = require("../middlewares/authenticate_user");
+
 // Controllers
 const controller = require("../controllers/user");
 
-router.use("/users", controller.get);
+// Routes
+const blogRouter = require("./blog");
+
+router.route("/").post(controller.create).get(controller.get);
+router.use("/:userId/blogs", authenticateUser, blogRouter);
+router
+  .route("/:userId")
+  .all(authenticateUser)
+  .get(controller.getOne)
+  .put(controller.update);
```

Comment:
@umairian if we see we need to authenticate the user for each protected route for that we are repeating `authenticateUser` mw in each route. It's best to use it only once and all the proceeding routes will go through it e.g:
```
router.all('/:userId/*,authenticateUser)
router.use("/:userId/blogs", blogRouter);
router
  .route("/:userId")
  .get(controller.getOne)
  .put(controller.update);
```